web/sharings/sharings.go

```diff
@@ -111,6 +111,37 @@ func GetSharing(c echo.Context) error {
 	return jsonapi.Data(c, http.StatusOK, as, nil)
 }
 
+// GetSharingsInfoByDocType returns
+func GetSharingsInfoByDocType(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+	docType := c.Param("doctype")
+
+	sharingIDs, err := sharing.GetSharingsByDocType(inst, docType)
+	if err != nil {
+		return wrapErrors(err)
+	}
+	sharings := make([]*sharing.APISharing, len(sharingIDs))
+
+	for i, sharingID := range sharingIDs {
+		s, err := sharing.FindSharing(inst, sharingID)
+		if err != nil {
+			return wrapErrors(err)
+		}
+		docs, err := sharing.GetSharedDocsBySharingID(inst, sharingID)
+		if err != nil {
+			return wrapErrors(err)
+		}
+		as := &sharing.APISharing{
+			Sharing:     s,
+			Credentials: nil,
+			SharedDocs:  docs,
+		}
+		sharings[i] = as
+	}
```

Comment:
We should avoid to put the same sharings several times in the response, and to query the sharings one by one. In go, we should use a map to deduplicate the sharing ids (`var seen map[string]interface{}`) and load all the sharings at once with a query on _all_docs with `keys: sharingsIDs`.